src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -775,48 +840,427 @@ private void populateRouting(Map<String, String> stats, String methodName, Strin
+	 *
+	 * @param screenName Name of Super Screen Destination
+	 * @param sourceName The source that will be assigned to the Super Screen Destination
+	 * @throws Exception when fail to get destination content
+	 */
+	private void changeSuperScreenContent(String screenName, String sourceName) throws Exception {
+		// STEP 1:  From superScreenDestination => get global layer map and list of screen destination
+		// Exception are thrown when there's no global layer & no collection of screen destination
+		SuperDestination superDestination = populateSuperDestination(screenName);
+		// STEP 2: From screen dest name list => get id of screen destination that has the largest area.
+		int currentScreenDestId = getDestHasMaxAreaId(superDestination);
+		// STEP 3: From screen dest id =>  getScreenDestContent
+		JsonNode currenScreenDestContent = getScreenDestContent(currentScreenDestId);
+		final JsonNode layerNode = currenScreenDestContent.get(BarcoE2Constant.LAYERS);
+		int globalLayerId = -1;
+		for (int i = 0; i < layerNode.size(); i++) {
+			JsonNode currentLayer = layerNode.get(i);
+			if (currentLayer != null && superDestination.getId() == currentLayer.get(BarcoE2Constant.LINK_DEST_ID).asInt()) {
+				globalLayerId = currentLayer.get(BarcoE2Constant.ID).asInt();
+				break;
+			}
+		}
+		boolean isSuccess;
+		int getSourceIndex = getNewSourceIndex(sourceName);
+		boolean isMixedLayer = checkLayerType(currentScreenDestId);
+		if (globalLayerId == -1) {
+			// CASE 1 assign to first layer
+			isSuccess = assignAndVerifySource(currentScreenDestId, getSourceIndex, isMixedLayer, 0);
+		} else {
+			// CASE 2 assign to globalLayerId's layer
+			isSuccess = assignAndVerifySource(currentScreenDestId, getSourceIndex, isMixedLayer, globalLayerId);
+			if (!isSuccess) {
+				// retry 1 more time on layer 0
+				 isSuccess = assignAndVerifySource(currentScreenDestId, getSourceIndex, isMixedLayer, 0);
+			}
+		}
+		if (isSuccess) return;
+		throw new CommandFailureException(this.getAddress(), "changeContent", String.format("Fail to call changeContent on super destination: %s", screenName));
+	}
+
+	/**
+	 * Assign a source to destination and verify if the data is changed.
+	 *
+	 * @param currentScreenDestId Screen destination id
+	 * @param getSourceIndex new source index that will be assigned
+	 * @param isMixedLayer is mixed layer or single layer
+	 * @param layerIndex index of layer
+	 * @return This returns boolean indicates that success/fail to assign source
+	 * @throws Exception Throws exception when fail call post in the device
+	 */
+	private boolean assignAndVerifySource(int currentScreenDestId, int getSourceIndex, boolean isMixedLayer, int layerIndex) throws Exception {
+		boolean isSuccess;
+		clearLayerFromDest(currentScreenDestId, layerIndex, isMixedLayer);
+		assignToDest(getSourceIndex, currentScreenDestId, layerIndex, isMixedLayer);
+		// Sleep here to wait for the device to assign a source.
+		Thread.sleep(1000);
```

Comment:
is it always enough to wait for 1s?

Reply:
Yes, I tested it. With the simulator, we only need about 100 ms. So I think 1000 ms is good enough for our case